lib/streamlit/elements/arrow_altair.py

```diff
@@ -608,6 +601,12 @@ def id_transform(data) -> Dict[str, str]:
 
     alt.data_transformers.register("id", id_transform)
 
+    if alt.themes.active == "default":
+        # The default altair theme has some width/height defaults defined
+        # that are not useful for Streamlit. Therefore, we change the theme to
+        # "none" to avoid those defaults.
+        alt.themes.enable("none")
```

Comment:
Minor remark but you could move the enabling of the "none" theme into the `with` block on line 610 so that the theme is reverted back to "default". Maybe a user relies on the default width/height settings when exporting the chart with a custom "Save as PNG" button in their app. Maybe with `nullcontext` as proposed [here](https://stackoverflow.com/a/34798330)?

Reply:
Thanks for the tip 👍  I changed it to use the context manager and nullcontext.